enterprise/server/remote_execution/executor/executor.go

```diff
@@ -68,11 +68,11 @@ type Options struct {
 func NewExecutor(env environment.Env, id string, runnerPool interfaces.RunnerPool, options *Options) (*Executor, error) {
 	hostID := options.NameOverride
 	if hostID == "" {
-		if h, err := uuid.GetHostID(); err == nil {
+		if h, err := uuid.GetHostID(env.GetFileCache().HostIDFilepath()); err == nil {
```

Comment:
It doesn't seem like this needs to be tied to the filecache at all, but maybe I'm missing some context. (Let me know if so!)

Instead I think this should be the "rootDirectory" configured in the executor config, and/or a different one for the app?

Reply:
I suggested putting it under the filecache dir because it was guaranteed to have the same lifecycle as the filecache - so if the filecache is persistent, the host_id will also be persistent. I think it's probably fine to put it in executor.root_directory though since that probably also has the same lifecycle as the filecache - otherwise things like hardlinking probably wouldn't work? (since different lifecycle is probably means different FS)

So +1 to putting it under executor.root_directory

Reply:
thanks, done!